config/http_config.go

```diff
@@ -766,9 +765,6 @@ func toSecret(secretManager SecretManager, text Secret, file, ref string) (secre
 		}, nil
 	}
 	if ref != "" {
-		if secretManager == nil {
```

Comment:
The problem with removing this is that you have to always call `WithSecretManager` or else this would crash, so this it not completely backwards compatible.

Reply:
ayaya, I wanted to remove it but then wanted to get it back, sorry. Ignore this.

Reply:
Fixed.